### ISSUE TITLE
How should I attach a generic onChange handler for array values?

### ISSUE THREAD

**Issue by scotttrinh** [Textalk/angular-schema-form#468]
I have a bunch of form fields of various types, including fixed-size arrays (tuples). I have to send the changes to a separate service for syncing, and until now, I've just been sending the whole object with a scope watch. This has worked, but I'm trying to refactor to only send the changed values, rather than the whole object. I've come up with a generic onChange handler that will handle normal fields, and if I attach them manually to the form definition, it seems to work. The issue is with arrays. How/where do I attach these onChange handlers so that if it gets called for all of the array values? I would be content with a change handler on the array itself that just gave me the new value of the whole array, if that's simpler.

**Comment by nicklasb** [comment]
Isn't ASFs own onChange triggered on arrays?

**Comment by scotttrinh** [comment]
It doesn't seem to be. Let me see if I can make a minimal plunker.

**Comment by scotttrinh** [comment]
[Plunker here](http://embed.plnkr.co/KU5QOC/preview)

**Comment by nicklasb** [comment]
Is this still a problem?

**Comment by scotttrinh** [comment]
Yes, as evidenced in the plunker. `console.log` fire when you change the string, but not when you change and array's value. That makes sense, honestly, since you're not changing the identity of the array, just the values, but I'm trying to solve this particular problem and looking for a way to do so.

**Comment by Anthropic** [comment]
@scotttrinh thanks for the update, good plunker thanks, I've tagged issue appropriately, it seems simple arrays are not firing as you demonstrated. I am curious to know if the new builder or broken out core will fix this, if not I will look into it.

**Comment by scotttrinh** [comment]
@Anthropic thanks for looking into it. I noticed that same bit of code when I was troubleshooting, which is what caused me to open this issue to see if there was a more "correct" way other people already use.
